monkey/common/agent_events/abstract_agent_event.py

```diff
@@ -26,5 +26,8 @@ class AbstractAgentEvent(InfectionMonkeyBaseModel, ABC):
 
     source: AgentID
     target: Union[IPv4Address, MachineID, None] = Field(default=None)
-    timestamp: float = Field(default_factory=time.time)
+    # The `lambda` is needed to mock `time.time()` in unit tests.
+    # See https://stackoverflow.com/a/61281048/.
+    # `pydantic.BaseModel` behaves like a dataclass.
```

Comment:
Why do we need to mock `time.time()` at all? Just use a constant for the timestamp in the test and pass a value to `TCPScanEvent's` constructor.